### ISSUE TITLE
[IMPROVEMENT] Network variable

### ISSUE THREAD

**Issue by Lennord** [Zulfen/ZulfBungee#12]
Hello, I would like to know if it is possible to make network variables faster (instantly direct), because network variables are a bit slow, it checks 1 variable per variable and so if I check several variables at the same time. time is quite slow.

**Comment by Lennord** [comment]
I tried changing the "packet-response-time" value to a lower number but it still works so slow.

**Comment by Zulfen** [comment]
Yeah, unfortunately due to limitations with Skript it makes it a bit hard to have decent performance with this addon. My code also has probably something to do with it though.

There might be a way around it using the new section stuff that has been added to Skript, but gotta figure out how to code that!

Until then, have you tried using `/zulfbungee ping` and setting the response time to something slightly higher than that? For example, I have mine set to 50 as my ping is around 46ms, and it works okay.

**Comment by Lennord** [comment]
> Yeah, unfortunately due to limitations with Skript it makes it a bit hard to have decent performance with this addon. My code also has probably something to do with it though.
> 
> There might be a way around it using the new section stuff that has been added to Skript that supports running the syntax in async, but gotta figure out how to code that!
> 
> Until then, have you tried using `/zulfbungee ping` and setting the response time to something slightly higher than that? For example, I have mine set to 50 as my ping is around 46ms, and it works okay.

Ok, I just tested the /zulfbungee ping and changed the config to the value I have in the ping but it still doesn't work, the network variables are still long.

In short, it does not matter, I will do it with time that you find a solution to solve this problem.

**Comment by MaitreFranck** [comment]
Based on my experience, using network/global variables isn't the best way to store players data on long term. If you want to store things like ranks, coins, ect you might be fine with databases like MongoDB or MySQL, with their related addons (I personnaly use MongoSK and it works fine). It can be a good solution for your speed issues.

**Comment by Lennord** [comment]
> 

Hello, Yes I agree, but the problem I have already tried to create a Mongodb database but the problem is that it is mandatory to protect it according to European law so I cannot protect it (I am French) but I have trouble with English.

**Comment by Lennord** [comment]
> Based on my experience, using network/global variables isn't the best way to store players data on long term. If you want to store things like ranks, coins, ect you might be fine with databases like MongoDB or MySQL, with their related addons (I personnaly use MongoSK and it works fine). It can be a good solution for your speed issues.

Hello, Yes I agree, but the problem I have already tried to create a Mongodb database but the problem is that it is mandatory to protect it according to European law so I cannot protect it (I am French) but I have trouble with English.

**Comment by Lennord** [comment]
Good evening, do you know if this will be changed in the next few days?

**Comment by Lennord** [comment]
Hello, is the variable loading time on the H2 the same as MySQL with your plugin?

**Comment by Lennord** [comment]
The network variable latency problem has been fixed with 0.9.0, thank you very much, I close the ticket.

**Comment by Lennord** [comment]
I noticed after closing the issue, that it works perfectly in locahost but not on a vps (Pterodactyl or pufferpanel), I'm sorry for closing the issue.

**Comment by Lennord** [comment]
Hi, sorry to ask you this question but it is to know if the network variables will be fixed on the vps including PufferPanel.

**Comment by Lennord** [comment]
Hello, I tested the variable networks on the "Release v0.9.8-pre1" and it is still slow.

I think you haven't fixed or changed this problem yet but I prefer to test when you changed a few things on the network variables.

Thank you for your understanding !

**Comment by Lennord** [comment]
I also noticed that if we check and modify several network variables it lowers the tps to around 13/20 tps.

**Comment by Lennord** [comment]
Good evening,

When I put in the configuration of the Proxy and spigot on "socket", it causes a latency of the network variables but when I put "pluginmessage", the network variables work very quickly but the problem is that I only want to put socket because with pluginmessage it must have a player connected for the network variables to work while I don't want.

Cordially.

INFO:
Zulfbungee version: ALPHA-0.9.8

**Comment by Lennord** [comment]
Good evening, I tested the latest build of the plugin and unfortunately there is still this problem of network variable latency.

**Comment by Lennord** [comment]
Hello @Zulfen, You wanted to correct this problem in a few days/weeks. Because it works on 'pluginmessage but it still doesn't work with "socket". Sorry to have pinged you. Have a nice day!

**Comment by Lennord** [comment]
Hello, sorry to bother you, do you have a date for the problem correction?

**Comment by Lennord** [comment]
@Zulfen

**Comment by Lennord** [comment]
@Zulfen
